Source/SwiftLintFramework/Rules/AccessControlOverrideOrderRule.swift

```diff
@@ -0,0 +1,114 @@
+        name: "Access Control Override Order",
+        description: "Access control property keywords should be followed by the override keyword.",
+        kind: .lint,
+        nonTriggeringExamples: [
+            "public override init()\n",
+            "internal override init()\n",
+            "private override init(){}\n",
+            "open override var foo: String\n",
+            "public override var foo: String\n",
+            "internal override var foo: String\n",
+            "private override var foo: String\n",
+            "open override func foo() -> String",
+            "public override func foo() -> String",
+            "internal override func foo() -> String",
+            "private override func foo() -> String"
+        ],
+        triggeringExamples: [
+            "↓override public init()\n",
+            "↓override internal init()\n",
+            "↓override private init(){}\n",
+            "↓override open var foo: String\n",
+            "↓override public var foo: String\n",
+            "↓override internal var foo: String\n",
+            "↓override private var foo: String\n",
+            "↓override open func foo() -> String",
+            "↓override public func foo() -> String",
+            "↓override internal func foo() -> String",
+            "↓override private func foo() -> String"
+        ])
+
+    public func validate(file: File, kind: SwiftDeclarationKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard
+            self.isOverride(attributes: dictionary.enclosedSwiftAttributes),
+            let accessControlLevel = dictionary.accessibility.flatMap(AccessControlLevel.init(identifier:)),
+            let keyOffset = dictionary.offset,
+            let syntaxTokens = self.syntaxTokens(at: keyOffset, in: file) else {
+                return []
+        }
+
+        let attributeTokens = self.attributeTokens(from: syntaxTokens)
+        let contents = self.textContents(of: attributeTokens, in: file)
+
+        guard
+            let accessIndex = contents.index(of: accessControlLevel.description),
+            let overrideIndex = contents.index(of: "override") else {
+                return []
+        }
+
+        if accessIndex > overrideIndex {
+            let offSet = self.safeOffSet(of: attributeTokens[overrideIndex], in: file)
+            return [StyleViolation(ruleDescription: type(of: self).description,
+                                   severity: .error,
+                                   location: Location(file: file, characterOffset: offSet),
+                                   reason: self.configuration.consoleDescription)]
+        }
+
+        return []
+    }
+
+    private func isOverride(attributes: [String]) -> Bool {
+        return attributes.contains("source.decl.attribute.override")
+    }
+
+    private func syntaxTokens(at keywordOffset: Int, in file: File) -> [SyntaxToken]? {
+        return file
```

Comment:
How about

```swift
    return file
        .syntaxTokensByLine()
        .flatMap { $0.first { $0.contains { $0.offset == keywordOffset } }
    }
}
```

Reply:
`.first` assumes a particular order. Is the order guaranteed?

Reply:
There should one, and only one what contains the given keyword, so .first is mostly used to get a single array.